snapcraft/tests/repo/test_snaps.py

```diff
@@ -0,0 +1,377 @@
+    ]
+
+    def test_is_classic(self):
+        snap_pkg = snaps.SnapPackage(self.snap)
+        self.assertThat(snap_pkg.is_classic(), Is(self.expected))
+
+
+class SnapPackageIsValidTest(SnapPackageBaseTestCase):
+
+    scenarios = [
+        ('valid',
+         dict(snap='fake-snap', expected=True)),
+        ('valid with channel',
+         dict(snap='fake-snap/strict/stable', expected=True)),
+        ('invalid',
+         dict(snap='missing-snap', expected=False)),
+        ('invalid with channel',
+         dict(snap='missing-snap/strict/stable', expected=False)),
+
+    ]
+
+    def test_is_valid(self):
+        snap_pkg = snaps.SnapPackage(self.snap)
+        self.assertThat(snap_pkg.is_valid(), Is(self.expected))
+
+    def test_is_valid_classmethod(self):
+        self.assertThat(snaps.SnapPackage.is_valid_snap(self.snap),
+                        Is(self.expected))
+
+
+class SnapPackageLifecycleTest(SnapPackageBaseTestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.fake_snap_command = FakeSnapCommand()
+        self.useFixture(self.fake_snap_command)
+
+    def test_install_classic(self):
+        snap_pkg = snaps.SnapPackage('fake-snap/classic/stable')
+        snap_pkg.install()
+        self.assertThat(self.fake_snap_command.calls, Equals([
+            ['snap', 'whoami'],
+            ['sudo', 'snap', 'install', 'fake-snap',
+             '--channel', 'classic/stable', '--classic']]))
+
+    def test_install_non_classic(self):
+        snap_pkg = snaps.SnapPackage('fake-snap/strict/stable')
+        snap_pkg.install()
+        self.assertThat(self.fake_snap_command.calls, Equals([
+            ['snap', 'whoami'],
+            ['sudo', 'snap', 'install', 'fake-snap',
+             '--channel', 'strict/stable']]))
+
+    def test_install_fails(self):
+        self.fake_snap_command.install_success = False
+        snap_pkg = snaps.SnapPackage('fake-snap/strict/stable')
+        self.assertRaises(errors.SnapInstallError, snap_pkg.install)
+
+    def test_refresh(self):
+        snap_pkg = snaps.SnapPackage('fake-snap/strict/stable')
+        snap_pkg.refresh()
+        self.assertThat(self.fake_snap_command.calls, Equals([
+            ['snap', 'whoami'],
+            ['sudo', 'snap', 'refresh', 'fake-snap',
+             '--channel', 'strict/stable']]))
+
+    def test_refresh_to_classic(self):
+        snap_pkg = snaps.SnapPackage('fake-snap/classic/stable')
+        snap_pkg.refresh()
+        self.assertThat(self.fake_snap_command.calls, Equals([
+            ['snap', 'whoami'],
+            ['sudo', 'snap', 'refresh', 'fake-snap',
+             '--channel', 'classic/stable', '--classic']]))
+
+    def test_refresh_fails(self):
+        snap_pkg = snaps.SnapPackage('fake-snap/strict/stable')
+        self.fake_snap_command.refresh_success = False
+        self.assertRaises(errors.SnapRefreshError, snap_pkg.refresh)
+
+    def test_bundle_install(self):
```

Comment:
Nit: the term "bundle" confused me. I think `test_install_multiple` would be much clearer.